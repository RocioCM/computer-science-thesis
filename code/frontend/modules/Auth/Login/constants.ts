import { FormBuilderField, INPUT_TYPES } from '@/common/hooks/useForm/types';
import * as validators from '@/common/hooks/useForm/validators';

export const LOGIN_FORM_STRUCT = [
  {
    name: 'email',
    required: true,
    default: '',
    validators: validators.isEmail,
  },
  {
    name: 'password',
    required: true,
    default: '',
  },
];

export const LOGIN_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'email',
    placeholder: '',
    label: 'Correo electrónico',
  },
  {
    type: INPUT_TYPES.password,
    name: 'password',
    placeholder: '',
    label: 'Contraseña',
  },
];
