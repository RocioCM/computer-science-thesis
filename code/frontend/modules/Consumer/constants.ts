import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

export const SEARCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'search',
    required: false,
    default: '',
  },
];

export const SEARCH_FORM_INPUTS: FormBuilderField[] = [
  {
    name: 'search',
    type: INPUT_TYPES.search,
    label: 'Código de seguimiento',
    placeholder: 'ABC1234XYZ',
  },
];

// ------- Waste bottle recycle form ------- //

export const RECYCLE_FORM_STRUCT: FieldData[] = [
  {
    name: 'trackingCode',
    required: true,
    default: '',
  },
  {
    name: 'recyclerUid',
    required: true,
    default: '',
  },
];

export const RECYCLE_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'trackingCode',
    placeholder: 'Ingresá el código de seguimiento',
    label: 'Código de seguimiento',
  },
  {
    type: INPUT_TYPES.autocomplete,
    name: 'recyclerUid',
    placeholder: 'Ingresá el nombre o correo del reciclador',
    label: 'Reciclador/a',
  },
];
