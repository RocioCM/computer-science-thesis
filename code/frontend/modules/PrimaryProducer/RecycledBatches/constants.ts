import { arrayOf, validators } from '@/common/hooks/useForm';
import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

export const SEARCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'search',
    required: false,
    default: '',
  },
];

export const SEARCH_FORM_INPUTS: FormBuilderField[] = [
  {
    name: 'search',
    type: INPUT_TYPES.text,
    label: 'Código de seguimiento',
    placeholder: 'ABC1234XYZ',
  },
];

// ----------- Recycling Batch Form ------------- //

export const BATCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'weight',
    required: true,
    default: '',
  },
  {
    name: 'size',
    required: true,
    default: '',
  },
  {
    name: 'materialType',
    required: true,
    default: '',
  },
  {
    name: 'composition',
    required: true,
    default: [{ name: '', amount: '', measureUnit: '' }],
    structure: arrayOf([
      {
        name: 'name',
        required: true,
        default: '',
      },
      {
        name: 'amount',
        required: true,
        default: '',
        validators: validators.isPositiveNumber,
      },
      {
        name: 'measureUnit',
        required: true,
        default: '',
      },
    ]),
  },
  {
    name: 'extraInfo',
    required: false,
    default: '',
  },
  {
    name: 'wasteBottleIds',
    required: false,
    default: [],
  },
  {
    name: 'createdAt',
    required: true,
    disabled: (form) => !!form.id,
    default: new Date().toISOString().split('T')[0],
  },
];

export const BATCH_FORM_INPUTS_1: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'weight',
    placeholder: 'Ingresá el peso del lote de material',
    label: 'Peso del fardo (en kg)',
  },
  {
    type: INPUT_TYPES.text,
    name: 'size',
    placeholder: 'Ingresá el tamaño del fardo',
    label: 'Tamaño del fardo (con unidades)',
  },
  {
    type: INPUT_TYPES.text,
    name: 'materialType',
    placeholder: 'Ingresá el tipo de material del lote',
    label: 'Tipo de material',
  },
  {
    type: INPUT_TYPES.date,
    name: 'createdAt',
    placeholder: 'Seleccioná la fecha de producción',
    label: 'Fecha de producción',
    max: new Date().toISOString().split('T')[0],
  },
  {
    type: INPUT_TYPES.textarea,
    name: 'extraInfo',
    placeholder: 'Ingresá observaciones adicionales',
    label: 'Observaciones adicionales (opcional)',
    containerClassName: 'col-span-2',
  },
];

export const BATCH_COMPOSITION_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'name',
    placeholder: 'Ingresá el nombre del material',
    label: 'Nombre del material',
  },
  {
    type: INPUT_TYPES.number,
    name: 'amount',
    placeholder: 'Ingresá la cantidad de material',
    label: 'Cantidad de material',
  },
  {
    type: INPUT_TYPES.text,
    name: 'measureUnit',
    placeholder: 'Ingresá la unidad de medida de la cantidad de material',
    label: 'Unidad de medida',
  },
];

// ------- Sell form ------- //

export const SELL_FORM_STRUCT: FieldData[] = [
  {
    name: 'buyerUid',
    required: true,
    default: '',
  },
];

export const SELL_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.autocomplete,
    name: 'buyerUid',
    placeholder: 'Ingresá el nombre o correo del comprador',
    label: 'Comprador',
  },
];

//// TODO: remove unused
