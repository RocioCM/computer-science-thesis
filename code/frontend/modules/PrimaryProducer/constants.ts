import { arrayOf, object, validators } from '@/common/hooks/useForm';
import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

// type Material = {
//   name: string;
//   amount: number;
//   measureUnit: string;
// };

// type Bottle = {
//   weight: number; // uint256
//   color: string;
//   thickness: number; // uint256
//   composition: Material[];
//   shapeType: string;
//   originLocation: string;
//   extraInfo: string;
// };

// export type BottleBatch = {
//   id: number; // uint256
//   quantity: number; // uint256
//   bottleType: Bottle;
//   soldQuantity: number; // uint256
//   owner: string; // Ethereum address
//   createdAt: string; // Timestamp as a string
// };

export const BATCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'quantity',
    required: true,
    default: '',
    validators: validators.isNaturalNumber,
  },
  {
    name: 'bottleType',
    required: true,
    structure: object([
      {
        name: 'weight',
        required: true,
        default: '',
        validators: validators.isPositiveNumber,
      },
      {
        name: 'color',
        required: true,
        default: '',
      },
      {
        name: 'composition',
        required: true,
        default: [{ name: '', amount: '', measureUnit: '' }],
        structure: arrayOf([
          {
            name: 'name',
            required: true,
            default: '',
          },
          {
            name: 'amount',
            required: true,
            default: '',
            validators: validators.isPositiveNumber,
          },
          {
            name: 'measureUnit',
            required: true,
            default: '',
          },
        ]),
      },
      {
        name: 'thickness',
        required: true,
        default: '',
        validators: validators.isPositiveNumber,
      },
      {
        name: 'shapeType',
        required: true,
        default: '',
      },

      {
        name: 'originLocation',
        required: true,
        default: '',
      },
      {
        name: 'extraInfo',
        required: false,
        default: '',
      },
    ]),
  },
  {
    name: 'createdAt',
    required: true,
    disabled: (form) => !!form.id,
    default: new Date().toISOString().split('T')[0],
  },
];

export const BATCH_FORM_INPUTS_1: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'quantity',
    placeholder: 'Ingresá la cantidad de envases',
    label: 'Cantidad de envases',
  },
  {
    type: INPUT_TYPES.number,
    name: 'bottleType.weight',
    placeholder: 'Ingresá el peso por envase (gramos)',
    label: 'Peso por envase (en gramos)',
  },
  {
    type: INPUT_TYPES.dropdown,
    name: 'bottleType.color',
    placeholder: 'Seleccioná el color del envase',
    label: 'Color del envase',
    dropdownOptions: [
      { value: 'transparente', label: 'Transparente' },
      { value: 'verde', label: 'Verde' },
      { value: 'ambar', label: 'Ámbar' },
      { value: 'otro', label: 'Otro' },
    ],
  },
  {
    type: INPUT_TYPES.number,
    name: 'bottleType.thickness',
    placeholder: 'Ingresá el espesor (mm)',
    label: 'Espesor (en milímetros)',
  },
  {
    type: INPUT_TYPES.text,
    name: 'bottleType.shapeType',
    placeholder: 'Ingresá el tipo/forma del envase',
    label: 'Tipo/forma del envase',
  },
];

export const BATCH_COMPOSITION_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'name',
    placeholder: 'Ingresá el nombre del material',
    label: 'Nombre del material',
  },
  {
    type: INPUT_TYPES.number,
    name: 'amount',
    placeholder: 'Ingresá la cantidad de material',
    label: 'Cantidad de material',
  },
  {
    type: INPUT_TYPES.text,
    name: 'measureUnit',
    placeholder: 'Ingresá la unidad de medida de la cantidad de material',
    label: 'Unidad de medida',
  },
];

export const BATCH_FORM_INPUTS_2: FormBuilderField[] = [
  {
    type: INPUT_TYPES.date,
    name: 'createdAt',
    placeholder: 'Seleccioná la fecha de producción',
    label: 'Fecha de producción',
    max: new Date().toISOString().split('T')[0],
  },
  {
    type: INPUT_TYPES.text,
    name: 'bottleType.originLocation',
    placeholder: 'Ingresá la localización de producción',
    label: 'Localización de producción',
  },
  {
    type: INPUT_TYPES.textarea,
    name: 'bottleType.extraInfo',
    placeholder: 'Ingresá observaciones adicionales',
    label: 'Observaciones adicionales (opcional)',
    containerClassName: 'col-span-2',
  },
];

// ------- Sell form ------- //

export const SELL_FORM_STRUCT: FieldData[] = [
  {
    name: 'quantity',
    required: true,
    default: '',
    validators: validators.isNaturalNumber,
  },
  {
    name: 'buyerUid',
    required: true,
    default: '',
  },
];

export const SELL_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'quantity',
    placeholder: 'Ingresá la cantidad de envases a vender',
    label: 'Cantidad de envases a vender',
  },
  {
    type: INPUT_TYPES.autocomplete,
    name: 'buyerUid',
    placeholder: 'Ingresá el nombre o correo del comprador',
    label: 'Comprador',
  },
];

// ------- Recycle form ------- //

export const RECYCLE_FORM_STRUCT: FieldData[] = [
  {
    name: 'quantity',
    required: true,
    default: '',
    validators: validators.isNaturalNumber,
  },
];

export const RECYCLE_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'quantity',
    placeholder: 'Ingresá la cantidad de envases a reciclar',
    label: 'Cantidad de envases a reciclar',
  },
];
