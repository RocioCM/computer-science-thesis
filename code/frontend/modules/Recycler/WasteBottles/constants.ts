import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

export const TABS = {
  available: 'Disponibles',
  recycled: 'Recicladas',
};

// ------- Search form ------- //

export const SEARCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'search',
    required: false,
    default: '',
  },
];

export const SEARCH_FORM_INPUTS: FormBuilderField[] = [
  {
    name: 'search',
    type: INPUT_TYPES.text,
    label: 'CÃ³digo de seguimiento',
    placeholder: 'ABC1234XYZ',
  },
];

// ------- Recycle form ------- //

export const ASSIGN_FORM_STRUCT: FieldData[] = [
  {
    name: 'batchId',
    required: true,
    default: null,
  },
];

export const ASSIGN_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.dropdown,
    name: 'batchId',
    placeholder: 'Selecciona un lote',
    label: 'ID del lote',
  },
];
