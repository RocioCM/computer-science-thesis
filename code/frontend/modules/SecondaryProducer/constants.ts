import { validators } from '@/common/hooks/useForm';
import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

export const BATCH_FORM_STRUCT: FieldData[] = [
  {
    name: 'trackingCode',
    required: true,
    default: '',
  },
];

export const BATCH_CODE_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'trackingCode',
    placeholder: 'ABC1234XYZ',
    label: 'CÃ³digo de seguimiento',
  },
];

// ------- Sell form ------- //

export const SELL_FORM_STRUCT: FieldData[] = [
  {
    name: 'quantity',
    required: true,
    default: '',
    validators: validators.isNaturalNumber,
  },
  {
    name: 'buyerUid',
    required: true,
    default: '',
  },
];

export const SELL_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'quantity',
    placeholder: '100',
    label: 'Cantidad de botellas a vender',
  },
  {
    type: INPUT_TYPES.autocomplete,
    name: 'buyerUid',
    placeholder: 'Busca por nombre o correo del comprador',
    label: 'Comprador',
  },
];

// ------- Recycle form ------- //

export const RECYCLE_FORM_STRUCT: FieldData[] = [
  {
    name: 'quantity',
    required: true,
    default: '',
    validators: validators.isNaturalNumber,
  },
];

export const RECYCLE_FORM_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.number,
    name: 'quantity',
    placeholder: '100',
    label: 'Cantidad de envases a reciclar',
  },
];
