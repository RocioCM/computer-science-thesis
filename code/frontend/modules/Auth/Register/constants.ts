import { ROLES_OPTIONS } from '@/common/constants/auth';
import { FormBuilderField, INPUT_TYPES } from '@/common/hooks/useForm/types';
import * as validators from '@/common/hooks/useForm/validators';

export const REGISTER_FORM_STRUCT = [
  {
    name: 'email',
    required: true,
    default: '',
    validators: validators.isEmail,
  },
  {
    name: 'password',
    required: true,
    default: '',
    validators: validators.isPassword,
  },
  {
    name: 'roleId',
    required: true,
    default: null,
  },
];

export const REGISTER_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'email',
    placeholder: 'Ingresá tu email',
    label: 'Correo electrónico',
  },
  {
    type: INPUT_TYPES.password,
    name: 'password',
    placeholder: 'Ingresá tu contraseña',
    label: 'Contraseña',
  },
  {
    type: INPUT_TYPES.dropdown,
    name: 'roleId',
    placeholder: 'Seleccioná tu rol',
    label: 'Rol',
    dropdownOptions: ROLES_OPTIONS,
  },
];
