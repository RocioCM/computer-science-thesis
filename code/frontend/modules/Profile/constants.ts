import { ROLES_OPTIONS } from '@/common/constants/auth';
import { validators } from '@/common/hooks/useForm';
import {
  FieldData,
  FormBuilderField,
  INPUT_TYPES,
} from '@/common/hooks/useForm/types';

export const PROFILE_FORM_STRUCT: FieldData[] = [
  {
    name: 'email',
    disabled: true,
    default: '',
  },

  {
    name: 'role',
    disabled: true,
    default: null,
  },
  {
    name: 'blockchainId',
    disabled: true,
    default: '',
  },
  {
    name: 'userName',
    default: '',
  },
  {
    name: 'managerName',
    default: '',
  },
  {
    name: 'phone',
    default: '',
    validators: [validators.isPhone],
  },
];

export const PROFILE_INPUTS: FormBuilderField[] = [
  {
    type: INPUT_TYPES.text,
    name: 'email',
    label: 'Correo electrónico',
  },
  {
    type: INPUT_TYPES.dropdown,
    name: 'role',
    label: 'Rol',
    dropdownOptions: ROLES_OPTIONS,
  },
  {
    type: INPUT_TYPES.text,
    name: 'blockchainId',
    label: 'Dirección Blockchain',
  },
  {
    type: INPUT_TYPES.text,
    name: 'userName',
    label: 'Nombre de usuario',
  },
  {
    type: INPUT_TYPES.text,
    name: 'managerName',
    label: 'Nombre del Responsable',
  },
  {
    type: INPUT_TYPES.text,
    name: 'phone',
    label: 'Teléfono',
  },
];
