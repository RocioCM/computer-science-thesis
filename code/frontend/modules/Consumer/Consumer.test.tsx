import TestAppWrapper from '@/common/utils/tests';
import Page from '@/pages/consumer/index';
import { render, screen, fireEvent, act } from '@testing-library/react';
import { toast } from 'react-toastify';
import auth from '@/common/libraries/auth';
import { ROLES } from '@/common/constants/auth';

describe('Consumer Page', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Matches snapshot', () => {
    const { container } = render(
      <TestAppWrapper>
        <Page />
      </TestAppWrapper>
    );
    expect(container).toMatchSnapshot();
  });
});
