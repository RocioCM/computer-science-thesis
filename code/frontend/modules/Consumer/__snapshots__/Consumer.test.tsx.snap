// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consumer Page Matches snapshot 1`] = `
<div>
  <main
    class="w-full h-screen flex flex-col p-2xl"
  >
    <header
      class="pb-2xl flex items-center justify-between"
    >
      <h1>
        Botellas recicladas
      </h1>
      <button
        class="relative flex items-center justify-center gap-s rounded-rs font-semibold h-10 disabled:cursor-default disabled:bg-opacity-40  focusButton bg-p1 text-n0 py-s px-m w-max"
        data-testid="button"
      >
        <i
          class="fa-solid fa-search"
          data-testid="fa-icon"
        />
        Buscar
      </button>
    </header>
    <div
      class="w-full h-full overflow-auto"
      data-testid="table"
    >
      <table
        class="w-full relative h-fit"
      >
        <thead
          class="sticky top-0 z-10 bg-n0 rounded-lg text-left"
        >
          <tr>
            <th
              class="h6 p-s bg-p3 text-n0 last:text-center"
            >
              <span
                class="font-medium"
              >
                ID
              </span>
            </th>
            <th
              class="h6 p-s bg-p3 text-n0 last:text-center"
            >
              <span
                class="font-medium"
              >
                Código de seguimiento
              </span>
            </th>
            <th
              class="h6 p-s bg-p3 text-n0 last:text-center"
            >
              <span
                class="font-medium"
              >
                Fecha de creación
              </span>
            </th>
            <th
              class="h6 p-s bg-p3 text-n0 last:text-center"
            >
              <span
                class="font-medium"
              >
                Estado
              </span>
            </th>
            <th
              class="h6 p-s bg-p3 text-n0 last:text-center"
            >
              <span
                class="font-medium"
              >
                Acciones
              </span>
            </th>
          </tr>
        </thead>
        <tbody />
        <tfoot
          class="h-full"
        >
          <tr
            class="h-full"
          >
            <td
              class="px-4 py-2 h-full text-center"
              colspan="5"
            >
              <span
                class="block shrink-0 animate-spin w-8 text-[2rem] mx-auto mt-8"
                data-testid="loading-spinner"
                style="width: 1.7rem; font-size: 1.7rem;"
              >
                <i
                  class="fa-solid fa-recycle"
                  data-testid="fa-icon"
                />
              </span>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </main>
</div>
`;
